homeassistant/components/somfy/cover.py

```diff
@@ -114,3 +159,18 @@ def close_cover_tilt(self, **kwargs):
     def stop_cover_tilt(self, **kwargs):
         """Stop the cover."""
         self.cover.stop()
+
+    async def async_added_to_hass(self):
+        """Complete the initialization."""
+        await super().async_added_to_hass()
+        if self.optimistic:
+            # Restore the last state if we use optimistic
+            last_state = await self.async_get_last_state()
+
+            if last_state is not None and last_state.state in (
+                STATE_OPEN,
+                STATE_CLOSED,
+            ):
+                self._closed = last_state.state == STATE_CLOSED
+
+        await self.async_update()
```

Comment:
Why this? Can't we just add `True` to `async_add_entities` instead  ?

Reply:
Yes, but moot since https://github.com/home-assistant/core/pull/42434 is about to delete it :)